### PR TITLE
[ENH] Add v2 interface support for N-BEATS model

### LINKED ISSUE #2034: [ENH] Add v2 interface support for N-BEATS
Referencing #1992 

This issue tracks adding v2 interface support for the N-BEATS model.

The model currently does not have a v2 package. The plan is to interface the existing implementation following the v2 pattern used by DLinear, SAMformer, and TFT.

CC: @fkiraly

### PR DESCRIPTION
**Pull request by harshsomankar123-tech**
#### Reference Issues/PRs
Fixes #2034.

#### What does this implement/fix? Explain your changes.
This PR adds v2 interface support for the N-BEATS model, following the v2 patterns recently established by DLinear, SAMformer, and TFT in the codebase. 

Specifically, this implements:
* **Model Core (`pytorch_forecasting/models/nbeats/_nbeats_v2.py`):** * Implemented `NBeats` inheriting from the experimental `BaseModel` v2.
  * Added dynamic parameter metadata extraction for lookback (`context_length`) and horizon (`prediction_length`).
  * Implemented the forward pass to return prediction, backcast, trend, seasonality, and generic forecasts.
  * Squeezed target/prediction shapes where necessary to maintain strict compatibility with 2D loss metrics (e.g., MAE, QuantileLoss, MASE) and v2 training loops.
  * Integrated an optional backcast loss scaling hook when `backcast_loss_ratio > 0`.
  * Preserved model interpretation functionality (`plot_interpretation`) and scaling transformation (`transform_output`).
* **Package Wrapper & Registration (`pytorch_forecasting/models/nbeats/_nbeats_pkg_v2.py` & `__init__.py`):** * Created package container `NBeats_pkg_v2` inheriting from `Base_pkg` to represent the high-level workflow wrapper.
  * Registered model parameters, metadata, and testing configurations (`get_test_train_params`).
  * Updated imports and exports for the new v2 components.

#### What should a reviewer concentrate their feedback on?
* [x] The implementation of the target/prediction shape squeezing to ensure robust compatibility with 2D loss metrics.
* [x] The integration strategy with the new `BaseModel` (v2) and `Base_pkg` classes.
* [x] The optional backcast loss scaling hook logic.
* [x] Verification that the new `get_test_train_params` adequately covers required testing configurations for the package wrapper.

#### Did you add any tests for the change?
Yes. Robust test coverage was added in `tests/test_models/test_nbeats_v2.py`. 

The new tests cover initialization, forward pass configurations, quantile loss output shapes, step functions (training, validation, test), packaging/fitting pipelines, and interpretation plotting (including matplotlib soft-dependency checking). 

**Verification:** All new v2 tests (7 passes, 1 skipped due to matplotlib missing) and legacy v1 tests (2 passes, 1 skipped) execute successfully without regressions.

#### Any other comments?
N/A

#### PR checklist
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. 
- [x] Added/modified tests
- [x] Used pre-commit hooks when committing to ensure that code is compliant with hooks. Install hooks with `pre-commit install`.
  To run hooks independent of commit, execute `pre-commit run --all-files`

### CODE REVIEW

**Comment by harshsomankar123-tech** [comment]
@phoeenniixx @fkiraly Please take a look

File: Dockerfile
Comment (phoeenniixx) [inline Dockerfile:1]: why do we have this docker file?

File: pytorch_forecasting/models/nbeats/_nbeats_pkg_v2.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+from pytorch_forecasting.base._base_pkg import Base_pkg
+
+
+class NBeats_pkg_v2(Base_pkg):
+    """NBeats package container."""
+
+    _tags = {
+        "info:name": "NBeats",
+        "authors": ["jdb78"],
```
Comment (phoeenniixx) [inline pytorch_forecasting/models/nbeats/_nbeats_pkg_v2.py]: please add your name as well.

File: pytorch_forecasting/models/nbeats/_nbeats_v2.py [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+from pytorch_forecasting.layers._nbeats._blocks import (
+    NBEATSGenericBlock,
+    NBEATSSeasonalBlock,
+    NBEATSTrendBlock,
+)
+from pytorch_forecasting.metrics import MASE, Metric
+from pytorch_forecasting.models.base._base_model_v2 import BaseModel
+
+
+class NBeats(BaseModel):
```
Comment (phoeenniixx) [inline pytorch_forecasting/models/nbeats/_nbeats_v2.py]: Maybe name it `NBeats_v2` to create a distinction between v1 and v2?

**Review by phoeenniixx — CHANGES_REQUESTED** [review]
Please add the docstrings

File: Dockerfile
Comment (harshsomankar123-tech) [reply Dockerfile:1]: Sorry, I accidentally pushed that file.
